pkg/s3-proxy/server/tls_integration_test.go

```diff
@@ -0,0 +1,207 @@
+//go:build integration
+
+package server
+
+import (
+	"context"
+	"crypto/tls"
+	"io"
+	"net/http"
+	"testing"
+
+	"github.com/golang/mock/gomock"
+	"github.com/oxyno-zeta/s3-proxy/pkg/s3-proxy/config"
+	cmocks "github.com/oxyno-zeta/s3-proxy/pkg/s3-proxy/config/mocks"
+	"github.com/oxyno-zeta/s3-proxy/pkg/s3-proxy/log"
+	"github.com/oxyno-zeta/s3-proxy/pkg/s3-proxy/s3client"
+	"github.com/oxyno-zeta/s3-proxy/pkg/s3-proxy/tracing"
+	"github.com/oxyno-zeta/s3-proxy/pkg/s3-proxy/webhook"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestTLSServer(t *testing.T) {
```

Comment:
Can you add tests for certificates URL with HTTP server, s3, and a temporary file?
The others are untestable in CI without a way to mock those services but those can be done no?
Thanks!

Reply:
Ayup! Wasn't quite sure whether you prefer this to be in a unit or integration test. (It would be self-contained, not requiring the external containers, so it doesn't *have* to be an integration test. Wasn't sure what your practice was here; we generally try to put as much as will run into a unit test, but that's us...)

Reply:
This can be done in integration tests I think. As the tests create HTTP servers, fake s3 servers, ... I think this is a good place. That's where I put those tests in this project :) .